### PR TITLE
feat: search table by name and show unlisted tables

### PR DESCRIPTION
**Pull request by nathaliabj**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #817 #997 

### Description:
This PR adds the possibility to search tables and filter by unlisted tables, it also re organises some of the explorer components on their own files and introduces styled components to them.

### Preview:
Screenshot 2022-01-21 at 13 30 57

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by nathaliabj** [comment]
@PriPatel Any thoughts on the UI of the above?

**Review by ZeRego — APPROVED** [review]
lgtm 🎉 
Let's just wait for design feedback as well

**Comment by ZeRego** [comment]
@nathaliabj while testing this branch I'm getting an error ( sometimes ) when I go to the tables page. Not sure if it is related to your changes or another recent change. 
Screenshot 2022-01-24 at 09 52 22

**Comment by PriPatel** [comment]
@nathaliabj  @ZeRego  Looks good! My only suggestion is that the `See unlisted tables` toggle live under the searchbar as opposed to above. I think it would read more naturally that way - what do you all think?

**Comment by nathaliabj** [comment]
> @nathaliabj while testing this branch I'm getting an error ( sometimes ) when I go to the tables page. Not sure if it is related to your changes or another recent change. Screenshot 2022-01-24 at 09 52 22

@ZeRego I can't reproduce the error above, could you please try to run it again and let me know if it is still happening ?

**Comment by owlas** [comment]
@PriPatel what is an "unlisted table" ?

**Comment by ZeRego** [comment]
> @PriPatel what is an "unlisted table" ?

@owlas They are the tables that were filtered out. 
More details in https://github.com/lightdash/lightdash/issues/817

File: packages/frontend/src/components/Explorer/ExploreSideBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    const { errorLogs } = useApp();
+    const [search, setSearch] = useState<string>('');
+    const [filterExplores, toggleFilterExplores] = useToggle(true);
+    const exploresResult = useExplores(filterExplores);
+
+    const filteredTables = useMemo(() => {
+        if (exploresResult.data) {
+            if (search !== '') {
+                return new Fuse(Object.values(exploresResult.data), {
+                    keys: ['name', 'description'],
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreSideBar/index.tsx:56]: @nathaliabj we should remove `description` - we had feedback before that it made search quite poor. Alternatively we could weight the name much higher than the description. [docs on weighted search](https://fusejs.io/examples.html#weighted-search)

```
const fuse = new Fuse(books, {
  keys: [
    'description', // will be assigned a `weight` of 1
    {
      name: 'name',
      weight: 5
    }
  ]
})
```

**Comment by owlas** [comment]
@ZeRego @nathaliabj @PriPatel - nice! So these are the tables defaulted to not be shown in the sidebar.

In this case I think the switch should be default "off" (since the purpose of filtering tables is to hide by default) and to rename it "Show hidden tables"?

I was a bit confused by "unlisted"

File: packages/frontend/src/components/Explorer/ExploreSideBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                                        icon="cross"
+                                        onClick={() => setSearch('')}
+                                    />
+                                }
+                                placeholder="Search tables"
+                                value={search}
+                                onChange={(e) => setSearch(e.target.value)}
+                            />
+                        </SearchWrapper>
+                        <SwitchFilter
```
Comment (owlas) [inline packages/frontend/src/components/Explorer/ExploreSideBar/index.tsx:92]: Should this be inverted?
Screenshot 2022-01-24 at 13 45 22
Screenshot 2022-01-24 at 13 45 30

**Review by owlas — DISMISSED** [review]
Two comments.

Also I'm getting a crash. To reproduce:
1. Go to project settings
2. Change the models filter
3. Click save
4. Click on Explore > Tables [crashes]
Screenshot 2022-01-24 at 13 52 38

**Comment by nathaliabj** [comment]
https://www.loom.com/share/882592d04d994d82ba7dac56004c2de1

**Review by ZeRego — APPROVED** [review]
lgtm

**Pull request merged**